### ISSUE TITLE
[Bug] Upgrade 3.0.1PE->3.1.0PE fails

### ISSUE THREAD

**Issue by MrZANE42** [thingsboard/thingsboard#3300]
When trying to upgrade  3.0.1PE->3.1.0PE according to the instructions on https://thingsboard.io/docs/user-guide/install/pe/upgrade-instructions/#upgrading-to-31pe it seems to hangs on "Updating table event." after running
sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=3.0.1

I've waited for more than 40 minutes and not much happened. CPU usage is at <4% and nothing is going on on the SQL server.
RAM usage and disk usage is also very low, it seems like it's doing nothing at all.

Postgres server version is PostgreSQL 12.3 (Ubuntu 12.3-1.pgdg18.04+1) on x86_64-pc-linux-gnu.

Server Environment
* own setup
  * local infrastructure
  * 3.1.0PE
  * Linux ... 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Client Environment:
Not relevant

Expected behavior
Installer continues until finished.

Additional context:

Output of failed command:

sudo /usr/share/thingsboard/bin/install/upgrade.sh --fromVersion=3.0.1
 ===================================================
 :: ThingsBoard Professional Edition ::       (v3.1.0PE)
 ===================================================

Starting ThingsBoard Upgrade from version 3.0.1 ...
Upgrading ThingsBoard from version 3.0.1 to 3.1.0 ...
Updating schema ...
Optimizing alarm relations...
Alarm relations optimized.
Updating table admin_settings.
Table admin_settings column id updated!
Table admin_settings updated.
Updating table alarm.
Table alarm column id updated!
Table alarm column originator_id updated!
Table alarm column tenant_id updated!
Table alarm updated.
Updating table asset.
Table asset column id updated!
Table asset column customer_id updated!
Table asset column tenant_id updated!
Table asset updated.
Updating table audit_log.
Table audit_log column id updated!
Table audit_log column customer_id updated!
Table audit_log column tenant_id updated!
Table audit_log column entity_id updated!
Table audit_log column user_id updated!
Table audit_log updated.
Updating table attribute_kv.
Table attribute_kv column entity_id updated!
Table attribute_kv updated.
Updating table component_descriptor.
Table component_descriptor column id updated!
Table component_descriptor updated.
Updating table customer.
Table customer column id updated!
Table customer column tenant_id updated!
Table customer column parent_customer_id updated!
Table customer updated.
Updating table dashboard.
Table dashboard column id updated!
Table dashboard column tenant_id updated!
Table dashboard column customer_id updated!
Table dashboard updated.
Updating table device.
Table device column id updated!
Table device column customer_id updated!
Table device column tenant_id updated!
Table device updated.
Updating table device_credentials.
Table device_credentials column id updated!
constraint "device_credentials_device_id_unq_key" of relation "device_credentials" does not exist, skipping
Table device_credentials column device_id updated!
Table device_credentials updated.
Updating table event.  <--- stays here for a very long time

**Comment by CooL16** [comment]
It is okay since the whole schema of DB is changed.

**Comment by MrZANE42** [comment]
What do you mean by it's ok? 
The installer never seems to finish, should I press CTRL+C or just wait longer?
I'm sorry but I don't understand your answer.

**Comment by CooL16** [comment]
You have to wait longer then usual upgrade.
Please specify your PC specs as well @MrZANE42 .

**Comment by MrZANE42** [comment]
It's running on a Vmware instance with 8 vCPUs and 8GB assigned to it. The server is quite powerful but that doesn't help anything when it seems like it's doing nothing (No CPU,RAM or disk usage to speek of).
The event table is only 4309 rows so it really should take long.

**Comment by MrZANE42** [comment]
Sorry. I don't know why it at first try reported 4309 rows (timeout maybe?).
When i do it again it returns approx. 1.2 million rows, which ofcourse could take some time to alter.
But shouldn't the CPU and/or RAM usage go up during this?

**Comment by MrZANE42** [comment]
Can I just delete some of the older events (based on their timestamp) or will that corrupt things?

**Comment by CooL16** [comment]
Yes, you can delete data from table event.
It takes so long time because upgrade script migration data and updating DB schema

**Comment by MrZANE42** [comment]
Ok. But how come it's not using more CPU/RAM. This could go ALOT faster. CPU is around 4% of available for the instance.

**Comment by MrZANE42** [comment]
I think I found the reason. PGSQL wasn't buffering much in memory since it's default shared_buffer setting is very low. So it seems it might be disk speed limited. I will increase that and check that PGSQL caches correctly. This will increase the overall speed (considerably) also.

**Comment by MrZANE42** [comment]
It seems I need to read up on how the caching works on PGSQL (Never used it before starting using TB). Increaseing shared_buffers doesn't help much as it realies mostly on linux disk caching. 
Well, I think we can end this discussion here, I'll get back to you if pruning the events table doesn't help.
Thanks for your help

**Comment by chfxr** [comment]
We also saw this error but in a different manner - the database running out of disk space during the migration.
The following error occurs:
`org.postgresql.util.PSQLException: ERROR: could not extend file "base/24583/24676.6": No space left on device
  Hint: Check free disk space.
  Where: SQL statement "UPDATE event SET entity_id = to_uuid(old_entity_id) WHERE old_entity_id is not null;"`
  
 The original size was 5.6 GB, during the upgrade the database size increased over 10 GB.
 Our managed database got a size of 25GB, but the error still persists.
 After upgrading the database to 65GB, the error was gone and the migration succeed.

**Comment by flyback1010** [comment]
I too am stuck at "Updating table event." when trying to migrate the database from 2.4.2 to 3.3.2. Running thingsboard pe on AWS via EKS with Postgres DB managed by RDS. Executing 
`./k8s-upgrade-tb.sh --fromVersion=2.4.2` 
had been slowly but steadily working doing its thing until 3.0.1 to 3.1.0 migration but then stopped at "Updating table event". It had been 3 hours now without any change. CPU and Network bandwidth utilization of both database and Server where upgrade is actually executed remain minimal. 

Judging from the previous posts, the events table (which could indeed be quite big in our case) is proving a bottleneck for migration. Is there any official way of removing old events? Such a feature would make sense not only for migration but also for saving disk space and cleaning up after forgetting to disable debug mode.

Best regards
